### ISSUE TITLE
Enable compression for API Responses

### ISSUE THREAD

**Issue by frankinspace** [podaac/hydrocron#102]
See documentation: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-gzip-compression-decompression.html

Once we have benchmarking working (https://github.com/podaac/hydrocron/issues/22) we can experiment with a few different values for `minimumCompressionSize` to see what a reasonable value might be.

**Comment by nikki-t** [comment]
I have [enabled payload compression ](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-enable-compression.html#api-gateway-enable-compression-console) for the API in the SIT environment but I don't think it is returning a compressed response. 

I have tried:

- Enabling Content-Encoding and setting the minimum-compression-size to 0.
- Updated the GET method to set a header mapping in the integration response for the 200 response for `method.response.header.Access-Control-Allow-Headers` to `'Content-Type,Accept-Encoding'` and added the 200 response header `Access-Control-Allow-Headers` in the method response.
- Updated the OPTIONS integration response and method response with the same GET headers as above. 

I don't think I need to do the last two bullets for compression to work but thought I would try it.

According to this [documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-receive-response-with-compressed-payload.html), I need to send the header `Accept-Encoding:gzip` with the request to receive a GZIP-encoded payload. I believe all of the other conditions are satisfied.

I think that the Python requests library automatically decodes compressed responses so I tried the following curl request:

```bash
curl -vX --header 'Accept-encoding: gzip' --location --request GET 'https://soto.podaac.sit.earthdatacloud.nasa.gov/hydrocron/v1/timeseries?feature=Reach&feature_id=63470800171&start_time=2024-02-01T00:00:00%2b00:00&end_time=2024-10-30T00:00:00%2b00:00&output=csv&fields=reach_id,time_str,wse'
```

Which I think would return a GZIP-encoded response as I did not specify the `--compressed` argument but still no luck. @frankinspace - Any thoughts on what to try next?

**Comment by frankinspace** [comment]
@nikki-t I just looked in SIT and 'Content encoding' was inactive for the hydrocron API gateway. Perhaps your changes got overridden by another deployment? I manually updated the api to enable content encoding and set the compression size to 20480 Bytes. When I run the following request in PostMan it seems like the response is being compressed.
```
curl --location 'https://soto.podaac.sit.earthdatacloud.nasa.gov/hydrocron/v1/timeseries?feature=Reach&feature_id=72390300011&start_time=2024-01-01T00%3A00%3A00Z&end_time=2024-10-30T00%3A00%3A00Z&output=geojson&fields=reach_id%2Ctime_str%2Cwse%2Cgeometry'
```

When I send that request in PostMan I am able to view the 'Content-Encoding' header on the response and see that it is set to 'gzip'. When I remove the 'Accept-Encoding' header from my request (PostMan defaults to 'gzip, deflate, br'), the 'Content-Encoding' header is not returned on the response.

It seems to be behaving as expected when content encoding is enabled

**Comment by nikki-t** [comment]
@frankinspace - Yes I deployed `develop` to SIT to work on another issue which overwrote the compression changes I made. It's possible I was overthinking it and all of the testing I did was automatically decompressing the response!

I can update the Terraform to enable compression, do you want to start with a compression threshold of 20480 Bytes?

**Comment by frankinspace** [comment]
Sure, that particular query results in a response that is about 27kb so I just picked 20kb so I could see if it worked or not. I did a bit of googling on what a suggested minimum-compression-size is and there were suggestions that all seemed to land in the 10's of kb or less ballpark. We probably need to do some testing if we want to come up with a more accurate threshold but starting with 20kb seems fine.
